pkg/api/types.go

```diff
@@ -153,6 +153,15 @@ const (
 	TerminationMessagePathDefault string = "/dev/termination-log"
 )
 
+// There is no namespace required for these kinds specified here.
+// If a kind is not enumerated here, it is assumed to have a namespace scope.
+var NoneNamespacedKinds = []string{
```

Comment:
For clients, this should be via discovery and a RESTMapper.  RESTMapper has
all of this info today via registration.  We should probably be passing
RESTMapper into any code that makes decisions about generic object refs.

On Oct 23, 2015, at 12:36 PM, Tim Hockin <notifications@github.com> wrote:

In pkg/api/types.go
<https://github.com/kubernetes/kubernetes/pull/16146#discussion_r42885770>:

> @@ -153,6 +153,15 @@ const (
>  	TerminationMessagePathDefault string = "/dev/termination-log"
>  )
>
> +// There is no namespace required for these kinds specified here.
> +// If a kind is not enumerated here, it is assumed to have a namespace scope.
> +var NoneNamespacedKinds = []string{

The disadvantage of this is that the list is now far away from the place
where types are actually installed, which means it will probably get
forgotten.

@lavalamp <https://github.com/lavalamp> @smarterclayton
<https://github.com/smarterclayton> @derekwaynecarr
<https://github.com/derekwaynecarr> for ideas on how to do this best

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/16146/files#r42885770>.